src/FileService.Business/AddNewImageCommand.cs

```diff
@@ -14,34 +15,37 @@ public class AddNewImageCommand : IAddNewImageCommand
     {
         private readonly IImageRepository _repository;
         private readonly IValidator<ImageRequest> _validator;
-        private readonly IImageRequestMapper _mapper;
+        private readonly IDbImageMapper _mapper;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
         public AddNewImageCommand(
-             IImageRepository repository,
-             IValidator<ImageRequest> validator,
-             IImageRequestMapper mapper)
+            IImageRepository repository,
+            IValidator<ImageRequest> validator,
+            IDbImageMapper mapper,
+            IHttpContextAccessor httpContextAccessor)
         {
             _repository = repository;
             _validator = validator;
             _mapper = mapper;
+            _httpContextAccessor = httpContextAccessor;
         }
 
-        public Guid Execute(ImageRequest request)
+        public Guid Execute(ImageRequest request, Guid? userId = null)
         {
             _validator.ValidateAndThrowCustom(request);
 
-            var parentDbImage = _mapper.Map(request, ImageType.Full, out bool isBigImage);
+            Guid requiredUserId = userId ?? _httpContextAccessor.HttpContext.GetUserId();
+
+            var parentDbImage = _mapper.Map(request, ImageType.Full, out bool isBigImage, requiredUserId);
 
             if (isBigImage)
             {
-                var childDbImage = _mapper.Map(request, ImageType.Thumb, out isBigImage, parentDbImage.Id);
+                var childDbImage = _mapper.Map(request, ImageType.Thumb, out _, requiredUserId, parentDbImage.Id);
```

Comment:
"_" - It is so strange name , Why you use this name?

Reply:
it is not a name? it means that I don't need this after using method